// @sanity/studio-hints/lib/components/HintPage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _blockContentToReact = _interopRequireDefault(require("@sanity/block-content-to-react"));

var _arrowRight = _interopRequireDefault(require("part:@sanity/base/arrow-right"));

var _dynamic = _interopRequireDefault(require("next/dynamic"));

var _HintPage = _interopRequireDefault(require("./HintPage.css"));

var _internalLinkSerializer = _interopRequireDefault(require("./serializers/internalLinkSerializer"));

var _HintCard = _interopRequireDefault(require("./HintCard"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var YouTube = (0, _dynamic.default)(() => Promise.resolve().then(() => _interopRequireWildcard(require('./serializers/Video'))));
var Image = (0, _dynamic.default)(() => Promise.resolve().then(() => _interopRequireWildcard(require('./serializers/Image'))));
var CodeSnippet = (0, _dynamic.default)(() => Promise.resolve().then(() => _interopRequireWildcard(require('./serializers/CodeSnippet'))));
/**
 * This component isn't currently used,
 * but is saved for future iteration of the Studio Hints
 * Component probably needs more work.
 */

var serializers = {
  marks: {
    internalLink: _internalLinkSerializer.default,
    link: (_ref) => {
      var mark = _ref.mark,
          children = _ref.children;
      return /*#__PURE__*/_react.default.createElement("a", {
        href: mark.href,
        target: "_blank",
        rel: "noopener noreferrer"
      }, children);
    }
  },
  types: {
    youtube: (_ref2) => {
      var url = _ref2.node.url;
      return /*#__PURE__*/_react.default.createElement("div", {
        className: _HintPage.default.youtubeContainer
      }, /*#__PURE__*/_react.default.createElement(YouTube, {
        url: url,
        containerClassName: _HintPage.default.youtube,
        className: _HintPage.default.youtubeIframe
      }));
    },
    image: props => {
      return /*#__PURE__*/_react.default.createElement(Image, props);
    },
    code: (_ref3) => {
      var node = _ref3.node;
      var code = node.code,
          language = node.language,
          _key = node._key,
          highlightedLines = node.highlightedLines;

      if (!code) {
        console.error('empty code block');
        return /*#__PURE__*/_react.default.createElement("div", null);
      }

      return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(CodeSnippet, {
        key: _key,
        language: language == 'sh' ? 'bash' : language,
        value: code,
        markers: highlightedLines || []
      }));
    }
  }
};

function HintPage(props) {
  var hint = props.hint,
      onBackClick = props.onBackClick,
      nextHint = props.nextHint,
      onCardClick = props.onCardClick;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _HintPage.default.root
  }, /*#__PURE__*/_react.default.createElement("button", {
    className: _HintPage.default.backButton,
    onClick: () => onBackClick(null),
    type: "button"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _HintPage.default.inner,
    tabIndex: -1
  }, ' ', /*#__PURE__*/_react.default.createElement(_arrowRight.default, null), " Back")), /*#__PURE__*/_react.default.createElement("div", {
    className: _HintPage.default.blockContent
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: _HintPage.default.blockHeading
  }, hint.title), /*#__PURE__*/_react.default.createElement(_blockContentToReact.default, {
    blocks: hint.body || [],
    serializers: serializers
  })));
}

HintPage.propTypes = {
  hint: _propTypes.default.object.isRequired,
  nextHint: _propTypes.default.object.isRequired,
  onBackClick: _propTypes.default.func.isRequired,
  onCardClick: _propTypes.default.func.isRequired
};
var _default = HintPage;
exports.default = _default;
