// @sanity/studio-hints/lib/components/ToggleSidecarButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _iosHelpOutline = _interopRequireDefault(require("react-icons/lib/io/ios-help-outline"));

var _iosCloseOutline = _interopRequireDefault(require("react-icons/lib/io/ios-close-outline"));

var _sidecarDatastore = require("part:@sanity/default-layout/sidecar-datastore");

var _ToggleSidecarButton = _interopRequireDefault(require("./ToggleSidecarButton.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class ToggleSidecarButton extends _react.default.PureComponent {
  constructor() {
    super(...arguments);

    _defineProperty(this, "state", {
      isOpen: true
    });
  }

  componentDidMount() {
    this.subscription = _sidecarDatastore.isSidecarOpenSetting.listen().subscribe(isOpen => {
      this.setState({
        isOpen: isOpen !== false
      });
    });
  }

  componentWillUnmount() {
    if (this.subscription) {
      this.subscription.unsubscribe();
    }
  }

  render() {
    var isOpen = this.state.isOpen;
    return /*#__PURE__*/_react.default.createElement("button", {
      className: _ToggleSidecarButton.default.toggleButton,
      onClick: _sidecarDatastore.toggleSidecarOpenState,
      type: "button",
      "aria-label": "Toggle sidecar",
      "aria-pressed": isOpen
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: _ToggleSidecarButton.default.inner,
      tabIndex: -1
    }, isOpen ? /*#__PURE__*/_react.default.createElement(_iosCloseOutline.default, null) : /*#__PURE__*/_react.default.createElement(_iosHelpOutline.default, null)));
  }

}

exports.default = ToggleSidecarButton;
