// @sanity/studio-hints/lib/components/serializers/Image.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _imageUrl = _interopRequireDefault(require("@sanity/image-url"));

var _client = _interopRequireDefault(require("../../client"));

var _Image = _interopRequireDefault(require("./Image.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var builder = (0, _imageUrl.default)(_client.default);

function urlFor(source) {
  return builder.image(source).auto('format').fit('max').url();
}

function Image(props) {
  if (props.node.caption) {
    return /*#__PURE__*/_react.default.createElement("figure", {
      className: _Image.default.figure
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: _Image.default.imageContainer
    }, /*#__PURE__*/_react.default.createElement("img", {
      src: urlFor(props.node),
      alt: props.node.alt
    })), /*#__PURE__*/_react.default.createElement("figcaption", {
      className: _Image.default.figcaption
    }, props.node.caption));
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: _Image.default.imageContainer
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: urlFor(props.node),
    alt: props.node.alt
  }));
}

Image.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  node: _propTypes.default.object.isRequired
};
var _default = Image;
exports.default = _default;
